File: frontend/src/types/contentCluster.ts
export interface ContentCluster {
  id: number
  name: string
  description: string
  contentCount: number
  avgViews: number
  avgEngagement: number
  topContent: string
  tags: string[]
  platform: string
  createdAt: string
}

export interface ClusterContent {
  id: number
  title: string
  platform: string
  views: number
  likes: number
  engagement: number
  publishedAt: string
  similarity: number
}

export interface ContentClusterSummary {
  totalClusters: number
  totalContents: number
  avgClusterSize: number
  bestCluster: string
  mostCommonTag: string
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/contentcluster/ContentCluster.kt
package com.ongo.domain.contentcluster

import java.math.BigDecimal
import java.time.LocalDateTime

data class ContentCluster(
    val id: Long = 0,
    val workspaceId: Long,
    val name: String,
    val description: String? = null,
    val contentCount: Int = 0,
    val avgViews: Long = 0,
    val avgEngagement: BigDecimal = BigDecimal.ZERO,
    val topContent: String? = null,
    val tags: List<String> = emptyList(),
    val platform: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
)
